### ISSUE TITLE
Incorrect changes when update involves `r.filter`

### ISSUE THREAD

**Issue by simonratner** [neumino/reqlite#61]
#### Example 1 (append then filter):
```js
return r.table('T').insert({arr: []}, {returnChanges: 'always'}).then((res) => {
    // inserted: 1,
    // changes: [ {
    //   new_val: { arr: [], id: '7e8fbb6f-0683-d1db-0fe4-65b4efb7269c' },
    //   old_val: null
    // } ]
    return r.table('T').get(res.generated_keys[0]).update({
        arr: r.row('arr').append(1).filter((val) => true)
    }, {returnChanges: 'always'}).then((res) => {
        // unchanged: 1,
        // changes: [ {
        //   new_val: { arr: [1], id: '7e8fbb6f-0683-d1db-0fe4-65b4efb7269c' },
        //   old_val: { arr: [1], id: '7e8fbb6f-0683-d1db-0fe4-65b4efb7269c' }
        // } ]
    })
})
```
In this first case, the document is actually updated with the appended value, however the query result indicates that the document has not changed.

#### Example 2 (filter then append):
```js
return r.table('T').insert({arr: []}, {returnChanges: 'always'}).then((res) => {
    // inserted: 1,
    // changes: [ {
    //   new_val: { arr: [], id: '7e8fbb6f-0683-d1db-0fe4-65b4efb7269c' },
    //   old_val: null
    // } ]
    return r.table('T').get(res.generated_keys[0]).update({
        arr: r.row('arr').filter((val) => true).append(1)
    }, {returnChanges: 'always'}).then((res) => {
        // unchanged: 1,
        // changes: [ {
        //   new_val: { arr: [], id: '7e8fbb6f-0683-d1db-0fe4-65b4efb7269c' },
        //   old_val: { arr: [], id: '7e8fbb6f-0683-d1db-0fe4-65b4efb7269c' }
        // } ]
    })
})
```
In this case, the append() command does not take, and the document is not updated at all.

**Comment by neumino** [comment]
Thanks for the report, there are tests around that, so I'm not sure what's going on here. I'll try to take a look at it this week end.

**Comment by simonratner** [comment]
Thanks! Created some test cases for you.

**Comment by neumino** [comment]
@simonratner -- I really appreciate the pull request with the tests!
This issue was a bit tricky:
- We were not copying the value before appending a value
- We were not properly handling `sequences` in `util._merge` though with luck some tests with arrays were passing

It's fixed and released in 2.2.0
